pkg/logging/formatter.go

```diff
@@ -0,0 +1,164 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+
+	"golang.org/x/exp/slog"
+)
+
+type formatter struct {
+	l *slog.Logger
+
+	exiter   exiter
+	panicker panicker
+}
+
+type exiter interface {
+	Exit(code int)
+}
+
+type defaultExiter struct{}
+
+func (de defaultExiter) Exit(code int) {
+	os.Exit(code)
+}
+
+type panicker interface {
+	Panic(msg string)
+}
+
+type defaultPanicker struct{}
+
+func (dp defaultPanicker) Panic(msg string) {
+	panic(msg)
+}
+
+// Debugf allows writing of formatted debug messages to the logger
+func (f *formatter) Debugf(format string, args ...interface{}) {
+	if !f.l.Enabled(context.Background(), slog.LevelDebug) {
```

Comment:
I'd go out on a limb here and assume this won't perform because `context.Background()` is called each time only to check if something must be done. Why does `slog` require a context just to check what its log level is (or am I missing something)?